bin/libs/zos-dataset.ts

```diff
@@ -0,0 +1,293 @@
+
+/*
+  This program and the accompanying materials are made available
+  under the terms of the Eclipse Public License v2.0 which
+  accompanies this distribution, and is available at
+  https://www.eclipse.org/legal/epl-v20.html
+ 
+  SPDX-License-Identifier: EPL-2.0
+ 
+  Copyright Contributors to the Zowe Project.
+*/
+
+import * as std from 'std';
+
+import * as common from './common';
+import * as stringlib from './string';
+import * as shell from './shell';
+import * as zoslib from './zos';
+
+export function isDatasetExists(datasetName: string): boolean {
+  const result = shell.execSync('sh', '-c', `cat "//'${datasetName}'" 1>/dev/null 2>&1)`);
+  return result.rc === 0;
+}
+
+// Check if data set exists using TSO command (listds)
+// @return        0: exist
+//                1: data set is not in catalog
+//                2: data set member doesn't exist
+export function tsoIsDatasetExists(datasetName: string): number {
+  const result = zoslib.tsoCommand(`listds '${datasetName}' label`);
+  if (result.rc != 0) {
+    if (result.out.includes('NOT IN CATALOG')) {
+      return 1;
+    }
+
+    if (result.out.includes('MEMBER NAME NOT FOUND')) {
+      return 2;
+    }
+    // some other error we don't know yet
+    return 9;
+  }
+
+  return 0;
+}
+
+export function createDataSet(dsName: string, dsOptions: string): number {
+  const result=zoslib.tsoCommand(`ALLOCATE NEW DA('${dsName}') ${dsOptions}`);
+  return result.rc;
+}
+
+export function copyToDataset(filePath: string, dsName: string, cpOptions: string, allowOverwrite: boolean): number {
+  if (allowOverwrite != true) {
+    if (isDatasetExists(`//'${dsName}'`)) {
```

Comment:
this has `//` but so does the function it calls so it would have `////` which would be wrong...